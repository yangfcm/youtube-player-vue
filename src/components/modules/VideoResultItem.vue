<template>
  <div class="ui items">
    <div class="item">
      <router-link class="ui small image" :to="`/video/${video.id.videoId}`">
        <img :src="video.snippet.thumbnails.medium.url" />
      </router-link>
      <div class="content">
        <router-link class="header" :to="`/video/${video.id.videoId}`">
          {{
          video.snippet.title
          }}
        </router-link>
        <div class="meta">
          <router-link :to="`/channel/${video.snippet.channelId}`">{{ video.snippet.channelTitle }}</router-link>
          <div class="ui right floated">
            <i class="calendar plus icon"></i>
            {{ video.snippet.publishedAt | date }}
          </div>
        </div>
        <div class="description" style="overflow: auto;">{{ video.snippet.description }}</div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: ["video"],
};
</script>
<style scoped>
</style>