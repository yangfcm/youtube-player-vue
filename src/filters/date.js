/**
 * Convert a date object to human-readable string
 */
export default (date) => {
  const dateObj = new Date(date);
  const day = dateObj.getDate();
  const month = new Intl.DateTimeFormat("en-US", {
    month: "short",
  }).format(dateObj);
  const year = dateObj.getFullYear();
  return `${day} ${month}, ${year}`;
};
