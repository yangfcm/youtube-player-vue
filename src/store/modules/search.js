import axios from "../../settings";
import { SEARCH, CATCH_SEARCH_ERROR } from "../types";
import parseError from "../helpers/parseError";

const state = {
  searchResults: null,
  searchError: null,
};

const getters = {
  searchErrorMessage: (state) => {
    return parseError(state.searchError, "Failed to proceed searching");
  },
};

const mutations = {
  SEARCH: (state, payload) => {
    state.searchResults = payload;
    state.searchError = null;
  },
  CATCH_SEARCH_ERROR: (state, payload) => {
    state.searchError = payload;
  },
};

const actions = {
  search: async (context, keyword, pageToken) => {
    try {
      const response = await axios.get("/search", {
        params: {
          ...axios.defaults.params,
          q: keyword,
          part: "snippet",
          maxResults: 15,
          pageToken,
        },
      });
      context.commit(SEARCH, response.data);
    } catch (err) {
      context.commit(CATCH_SEARCH_ERROR, err);
    }
  },
};

export default {
  state,
  getters,
  mutations,
  actions,
};
