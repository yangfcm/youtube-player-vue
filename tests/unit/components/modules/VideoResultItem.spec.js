import { shallowMount, RouterLinkStub, createLocalVue } from "@vue/test-utils";
import VideoResultItem from "@/components/modules/VideoResultItem";
import { searchResultResponse } from "../../store/fixtures/search";
import date from "@/filters/date";

describe("Test VideoResultItem component", () => {
  let wrapper;
  const localVue = createLocalVue();
  localVue.filter("date", date);
  const video = searchResultResponse.items.find(
    (item) => item.id.kind === "youtube#video"
  );

  beforeEach(() => {
    wrapper = shallowMount(VideoResultItem, {
      propsData: {
        video,
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
      localVue,
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct content", () => {
    expect(wrapper.find("a.image").props().to).toBe(
      `/video/${video.id.videoId}`
    );
    expect(wrapper.find("a.header").props().to).toBe(
      `/video/${video.id.videoId}`
    );
    expect(wrapper.find("a.header").text()).toContain(video.snippet.title);
    expect(wrapper.find("div.meta a").props().to).toBe(
      `/channel/${video.snippet.channelId}`
    );
    expect(wrapper.find("div.meta a").text()).toContain(
      video.snippet.channelTitle
    );
    expect(wrapper.find(".description").text()).toContain(
      video.snippet.description
    );
  });
});
