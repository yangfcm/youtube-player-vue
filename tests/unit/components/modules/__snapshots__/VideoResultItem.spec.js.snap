// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VideoResultItem component should match snapshot 1`] = `
<div class="ui items">
  <div class="item"><a class="ui small image"><img src="https://i.ytimg.com/vi/H89hKw06iWs/mqdefault.jpg"></a>
    <div class="content"><a class="header">
        web.dev LIVE 2020: Day One
      </a>
      <div class="meta"><a>Google Chrome Developers</a>
        <div class="ui right floated"><i class="calendar plus icon"></i>
          1 Jul, 2020
        </div>
      </div>
      <div class="description" style="overflow: auto;">Join us for day 1 of web.dev LIVE as we head to the Americas timezone to provide helpful content on Performance and Discovery through Search. Enjoy the ...</div>
    </div>
  </div>
</div>
`;
